### PR TITLE
Deprecate the old 2d Draco_Mesh constructor

### PR DESCRIPTION
**Pull request by RyanWollaeger**
### Background

* The new dimension-independent `Draco_Mesh` constructor subsumes the fuctionality of the old 2D-only constructor.
* Removing the old 2D constructor simplifies the `Draco_Mesh` class and interface.
* However, the parser classes (X3D and RTT) must provide mesh data that is consistent with the newer mesh constructor.

### Purpose of Pull Request

* Deprecate the old 2d `Draco_Mesh` construction functions.
* [Fixes Redmine Issue #1654](https://rtt.lanl.gov/redmine/issues/1654)

### Description of changes

* Remove the older 2D-only `Layout` generation in `Draco_Mesh`.
* Update RTT parser to provide data compatible with the newer `Draco_Mesh` constructor.
* Remove logic in X3D parser that permits construction of older `Draco_Mesh` constructor arguments. 
* Add accessor to `face_type` vector in `Draco_Mesh`, which stores the number of nodes per face per cell.
* Update unit tests and testing apparatus for new `Draco_Mesh` version.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

File: src/mesh/Draco_Mesh.hh [hunk 1]
```diff
@@ -133,8 +134,14 @@ public:
   const std::vector<std::vector<double>> &get_node_coord_vec() const {
     return node_coord_vec;
   }
-  const std::vector<unsigned> &get_cell_type() const { return m_cell_type; }
-  const std::vector<unsigned> &get_cell_to_node_linkage() const {
+  const std::vector<unsigned> &get_num_faces_per_cell() const {
```
Comment (KineticTheory) [inline src/mesh/Draco_Mesh.hh]: Returning a `const&` to a class member can be a little risky.  If the class goes out of scope the `const&` will be invalid.  Is this really what you want here?

Comment (RyanWollaeger) [reply src/mesh/Draco_Mesh.hh]: That is a good question - the intent is to always have the class in scope, and to not copy this data when accessed, but this does seem risky.  Do you have any recommendations?

Comment (KineticTheory) [reply src/mesh/Draco_Mesh.hh]: I would prefer that you return a copy unless the memory cost is truly prohibitive.

Comment (RyanWollaeger) [reply src/mesh/Draco_Mesh.hh]: Will do - just going to make one other tweak before pushing.

**Comment by KineticTheory** [comment]
LGTM

**Comment by KineticTheory** [comment]
@jhchang-lanl Can you review/merge this PR?

**Pull request merged**